### ISSUE TITLE
Should offset be renamed index?

### ISSUE THREAD

**Issue by wbl** [wpack-wg/bundled-responses#10]
Offset makes me think that it's a distance in bytes. Index would be a little clearer that it's an index into the responses array. That's what was intended, right? Because if not, there's a fun bunch of security issues around smuggling.

**Comment by jyasskin** [comment]
It's actually a byte offset into the responses region, in order to support [random access](https://wicg.github.io/webpackage/draft-yasskin-wpack-use-cases.html#name-random-access) when the bundles are stored on disk. An index would require a linear scan.

That means that anyone (a virus scanner?) trying to check the responses for safety needs to start from the index, rather than only parsing the response array as a single CBOR object. I was focusing on clients that check for safety while loading, but it definitely makes sense to add a security consideration for folks trying to separate security checks from the actual use of the data.

We could also re-specify the responses section as a byte array instead of an array of CBOR objects, if you think that'd help.

**Comment by wbl** [comment]
There's no guarantee that the regions indicated are in order, disjoint, cover the contents etc? If everyone uses the index that's fine. I'll open a PR with some clarifying text and some warnings about common polyglots this enables. It probably can become a zip file for instance.

**Comment by jyasskin** [comment]
Right. I initially had text saying that anything that parsed was valid, and just advised against overlapping and holey responses, but someone encouraged me to remove that, in favor of it being invalid but unlikely to cause a parsing error. I'd especially appreciate text about how those polyglots become security vulnerabilities.

**Comment by wbl** [comment]
The usual way is signing: e.g. you thought you signed a webpack, but actually it's a zip with different contents, or a jpeg, etc.
